### PR TITLE
Rewrite id selector to be more standards-compliant

### PR DESCRIPTION
**Pull request by esdoppio**

### CODE REVIEW

File: grammars/css.cson [hunk 1]
```diff
@@ -500,18 +500,24 @@
...
         '''
         'name': 'entity.other.attribute-name.class.css'
       }
       {
         'captures':
           '1':
             'name': 'punctuation.definition.entity.css'
-        'match': '(#)[a-zA-Z][a-zA-Z0-9_-]*'
+        'match': '''(?x)
+          (\\#)
```
Comment (50Wliu) [inline grammars/css.cson]: No need for escape

File: spec/css-spec.coffee [hunk 2]
```diff
@@ -84,16 +84,41 @@ describe 'CSS grammar', ->
         expect(tokens[0]).toEqual value: '.- ', scopes: ['source.css', 'meta.selector.css']
 
     describe 'id selectors', ->
-      it 'tokenizes a standalone id selector', ->
-        {tokens} = grammar.tokenizeLine '#ch-1 {}'
+      it 'tokenizes \\#unicorn as id selector', ->
```
Comment (50Wliu) [inline spec/css-spec.coffee]: Surprise: I'm 99% sure that this test will still be skipped.  No matter how much escaping you do, Atom will still "helpfully" skip the test for you.

File: grammars/css.cson (hunk 1 above)
Comment (esdoppio) [inline grammars/css.cson]: It's true that `#` usually doesn't need escaping. But in extended mode (`?x`), `#` becomes a metacharacter, denoting the start of a comment—if you don't escape it, the following `)` will be interpreted as comment.

Comment (50Wliu) [inline grammars/css.cson]: right.

File: spec/css-spec.coffee (hunk 2 above)
Comment (esdoppio) [inline spec/css-spec.coffee]: Weird, but nothing is skipped in my case. You mean they will be skipped in the build?

Comment (50Wliu) [inline spec/css-spec.coffee]: I'll have to test this out later today.

Comment (50Wliu) [inline spec/css-spec.coffee]: Ok.  So the real problem is that if this fails, then the spec output is `it 'tokenizes \#unicorn as id selector'`.  Would you be ok with generalizing these, ie `it 'tokenizes common id selectors'`, `it 'tokenizes unicode id selectors'`, etc?

Comment (esdoppio) [inline spec/css-spec.coffee]: 👌

**Pull request merged**